test-utils/index.js

```diff
@@ -0,0 +1,222 @@
+                                    id: '1'
+                                },
+                                amount  : '1000000000000000000'
+                            }
+                        }
+                    ]
+                }
+            },
+            transfers  : {
+                single: '100',
+                total : '100'
+            }
+        };
+    }
+
+    async createSignedReceipt(unsignedReceipt) {
+        let signedReceipt = {
+            ...unsignedReceipt,
+            seals: {
+                ...unsignedReceipt.seals,
+                operator: {
+                    hash: ApiPayloadFactory.hashReceipt(unsignedReceipt)
+                }
+            }
+        };
+        signedReceipt.seals.operator.signature = await this.signAsOperator(signedReceipt.seals.operator.hash);
+        return signedReceipt;
+    }
+
+    static hashPayment(unsignedPayment) {
+        const amountCurrencyHash = hash(
+            unsignedPayment.amount,
+            unsignedPayment.currency.ct,
+            unsignedPayment.currency.id
+        );
+        const senderHash = hash(
+            unsignedPayment.sender.wallet,
+            unsignedPayment.sender.data
+        );
+        const recipientHash = hash(
+            unsignedPayment.recipient.wallet
+        );
+
+        return hash(amountCurrencyHash, senderHash, recipientHash);
+    }
+
+    static hashReceipt(unsignedReceipt) {
+        const walletSignatureHash = hash(
+            {type: 'uint8', value: unsignedReceipt.seals.wallet.signature.v},
+            unsignedReceipt.seals.wallet.signature.r,
+            unsignedReceipt.seals.wallet.signature.s
+        );
+        const senderHash = hash(
+            hash(unsignedReceipt.sender.nonce),
+            hash(
+                unsignedReceipt.sender.balances.current,
+                unsignedReceipt.sender.balances.previous
+            ),
+            hash(
+                unsignedReceipt.sender.fees.single.amount,
+                unsignedReceipt.sender.fees.single.currency.ct,
+                unsignedReceipt.sender.fees.single.currency.id
+            ),
+            ApiPayloadFactory.hashFeesTotal(unsignedReceipt.sender.fees.total)
+        );
+        const recipientHash = hash(
+            hash(unsignedReceipt.recipient.nonce),
+            hash(
+                unsignedReceipt.recipient.balances.current,
+                unsignedReceipt.recipient.balances.previous
+            ),
+            ApiPayloadFactory.hashFeesTotal(unsignedReceipt.recipient.fees.total)
+        );
+        const transfersHash = hash(
+            unsignedReceipt.transfers.single,
+            unsignedReceipt.transfers.total
+        );
+        const operatorHash = hash(unsignedReceipt.operator.data);
+        return hash(walletSignatureHash, senderHash, recipientHash, transfersHash, operatorHash);
+    }
+
+    static hashFeesTotal(totalFigures) {
+        let feesTotalHash = 0;
```

Comment:
Is it supposed to return `0` if you provide it an empty array? Is `0` the equivalent of hashing an empty array?

Reply:
No idea actually. This code was extracted from the testsuite and I dont think i changed it.